aalto/jupyterhub/nbgrader-jupyterlab.rst

```diff
@@ -0,0 +1,83 @@
+Doing Assignments in JupyterLab
+===============================
+
+.. note::
+
+    This is a tutorial for courses that distribute and collect
+    their assignments on `JupyterHub <https://jupyter.cs.aalto.fi>`__.
+    Feel free to use JupyterHub for your other courses as well, but
+    note that you won't be able to collect and submit your
+    assignments this way.
+
+Go to https://jupyter.cs.aalto.fi and sign in with your account to access JupyterHub.
+You can find more general information about it in :doc:`../jupyterhub`.
+Select your course and start a session. Your will see that you are under the ``notebooks/`` folder,
+which is your home directory. Make sure you put everything important here, files saved
+elsewhere will get deleted when your session ends.
+
+.. figure:: img/home.png
+   :align: center
+|
+You can see currently available assignments by going to "Nbgrader->Assignment List"
+in the menu bar.
+
+.. figure:: img/assignment_list.png
+   :align: center
+|
+If no assignments are visible, make sure the right course is selected
+in the drop down menu. It might also be possible that your instructor has
+not released any assignments yet.
+
+Click in the "Fetch" button to download the assignment into ``notebooks/<course-name>/``.
+
+.. figure:: img/fetch_before.png
+   :align: center
+.. figure:: img/fetch_after.png
+   :align: center
+|
+You can also click on the assignment name under "Downloaded assignments" to list
+and quickly open the associated notebooks.
+
+.. figure:: img/downloaded.png
+   :align: center
+|
+Follow the instructions in the notebook, and don't forget to save your work!
+Once you are done, you can go back to the Assignment List and press the "Submit" button.
+You can submit an assignment multiple times, but your instructor will probably use
+only your latest submission.
+
+At this point, you can delete the assignment files from your ``notebooks/`` directory if you'd like,
+but it is often a good idea to keep a copy of your work somewhere.
```

Comment:
It actually does cache a copy in `.local/share/jupyter` (or somewhere like that - I didn't check).  Should this be mentioned or is it needlessy confusing and detailed?  I don't think we mention it anywhere else, and the only major significance is if it gets too big and exceeds their quota...

Reply:
The cache affects functionality: if you submit an assignment and then remove it (e.g. from `notebooks/`), you can still fetch the feedback for it. In fact, you need the cache to fetch feedback at any point.

So I think we can either be explicit about its existence and tell users to not mess with it unless they understand the implications, or continue having obscurity and not worry about someone doing something wrong.

In an ideal world, I think the latter is better because everything that would take up a lot of space should be in `coursedata/` instead, so a few notebooks shouldn't create problems. In practice, I don't know if that is the case.